session/grpchijack/dial.go

```diff
@@ -46,6 +46,7 @@ type conn struct {
 
 	closedOnce sync.Once
 	readMu     sync.Mutex
+	writeMu    sync.Mutex
```

Comment:
I wonder why there was a readMu but not a writeMu in the first place, and if it's not better to have one RWMutex instead.